### DISCUSSION TITLE
Minikube Service pod error

### DISCUSSION THREAD

**Discussion by we-wake-act21** [eclipse-theia/theia-cloud#286] [Q&A]
Hi,

I'm currently attempting to deploy theia-cloud minikube on a Ubuntu - WSL instance on my local machine. The deployment of theia-cloud is successful, and I can access the Keycloak login page. However, upon login, I encounter an error with code 500.

Upon reviewing the logs from the service pod, I noticed the following:

> 2024-03-04 08:37:15,326 WARN  [io.qua.oid.run.OidcRecorder] (vert.x-eventloop-thread-10) OIDC server is not available at the 'https://127.0.0.1.nip.io/keycloak/realms/TheiaCloud' URL. Please make sure it is correct. Note it has to end with a realm value if you work with Keycloak, for example: 'https://localhost:8180/auth/realms/quarkus'
> 2024-03-04 08:37:15,326 ERROR [io.qua.ver.htt.run.QuarkusErrorHandler] (vert.x-eventloop-thread-10) HTTP Request to /service failed, error id: b56eda53-cb89-48ec-997f-0b49ebe03d37-2: io.quarkus.oidc.OIDCException: OIDC Server is not available
> at io.quarkus.oidc.runtime.OidcRecorder.toOidcException(OidcRecorder.java:337)
> 	at io.quarkus.oidc.runtime.OidcRecorder$10.apply(OidcRecorder.java:448)
> 	at io.quarkus.oidc.runtime.OidcRecorder$10.apply(OidcRecorder.java:442)
> 	at io.smallrye.context.impl.wrappers.SlowContextualBiFunction.apply(SlowContextualBiFunction.java:21)
> 	at io.smallrye.mutiny.operators.uni.UniOnItemOrFailureFlatMap$UniOnItemOrFailureFlatMapProcessor.performInnerSubscription(UniOnItemOrFailureFlatMap.java:86)
> 	at io.smallrye.mutiny.operators.uni.UniOnItemOrFailureFlatMap$UniOnItemOrFailureFlatMapProcessor.onFailure(UniOnItemOrFailureFlatMap.java:65)
> 	at io.smallrye.mutiny.operators.uni.UniOperatorProcessor.onFailure(UniOperatorProcessor.java:55)
> 	at io.smallrye.mutiny.operators.uni.UniOnFailureTransform$UniOnFailureTransformProcessor.onFailure(UniOnFailureTransform.java:64)
> 	at io.smallrye.mutiny.operators.uni.builders.UniCreateFromPublisher$PublisherSubscriber.onError(UniCreateFromPublisher.java:78)
> 	at io.smallrye.mutiny.helpers.HalfSerializer.onError(HalfSerializer.java:59)
> 	at io.smallrye.mutiny.helpers.StrictMultiSubscriber.onFailure(StrictMultiSubscriber.java:90)
> 	at io.smallrye.mutiny.subscription.MultiSubscriber.onError(MultiSubscriber.java:73)
> 	at io.smallrye.mutiny.subscription.SerializedSubscriber.onFailure(SerializedSubscriber.java:101)
> 	at io.smallrye.mutiny.operators.multi.MultiRetryWhenOp$RetryWhenOperator.whenFailure(MultiRetryWhenOp.java:165)
> 	at io.smallrye.mutiny.operators.multi.MultiRetryWhenOp$TriggerSubscriber.onError(MultiRetryWhenOp.java:193)
> 	at io.smallrye.mutiny.helpers.HalfSerializer.onError(HalfSerializer.java:59)
> 	at io.smallrye.mutiny.helpers.StrictMultiSubscriber.onFailure(StrictMultiSubscriber.java:90)
> 	at io.smallrye.mutiny.operators.multi.MultiConcatMapOp$ConcatMapMainSubscriber.postponeFailure(MultiConcatMapOp.java:238)
> 	at io.smallrye.mutiny.operators.multi.MultiConcatMapOp$ConcatMapMainSubscriber.innerFailure(MultiConcatMapOp.java:214)
> 	at io.smallrye.mutiny.operators.multi.MultiConcatMapOp$ConcatMapInner.onFailure(MultiConcatMapOp.java:297)
> 	at io.smallrye.mutiny.subscription.MultiSubscriber.onError(MultiSubscriber.java:73)
> 	at io.smallrye.mutiny.converters.uni.UniToMultiPublisher$UniToMultiSubscription.onFailure(UniToMultiPublisher.java:102)
> 	at io.smallrye.mutiny.operators.uni.builders.UniCreateFromKnownFailure$KnownFailureSubscription.forward(UniCreateFromKnownFailure.java:38)
> 	at io.smallrye.mutiny.operators.uni.builders.UniCreateFromKnownFailure.subscribe(UniCreateFromKnownFailure.java:23)
> 	at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)
> 	at io.smallrye.mutiny.converters.uni.UniToMultiPublisher$UniToMultiSubscription.request(UniToMultiPublisher.java:73)
> 	at io.smallrye.mutiny.subscription.SwitchableSubscriptionSubscriber.setOrSwitchUpstream(SwitchableSubscriptionSubscriber.java:205)
> 	at io.smallrye.mutiny.subscription.SwitchableSubscriptionSubscriber.onSubscribe(SwitchableSubscriptionSubscriber.java:107)
> 	at io.smallrye.mutiny.converters.uni.UniToMultiPublisher.subscribe(UniToMultiPublisher.java:25)
> 	at io.smallrye.mutiny.groups.MultiCreate$1.subscribe(MultiCreate.java:165)
> 	at io.smallrye.mutiny.operators.multi.MultiConcatMapOp$ConcatMapMainSubscriber.onItem(MultiConcatMapOp.java:144)
> 	at io.smallrye.mutiny.subscription.MultiSubscriber.onNext(MultiSubscriber.java:61)
> 	at io.smallrye.mutiny.operators.multi.processors.UnicastProcessor.drainWithDownstream(UnicastProcessor.java:107)
> 	at io.smallrye.mutiny.operators.multi.processors.UnicastProcessor.drain(UnicastProcessor.java:138)
> 	at io.smallrye.mutiny.operators.multi.processors.UnicastProcessor.onNext(UnicastProcessor.java:204)
> 	at io.smallrye.mutiny.operators.multi.processors.SerializedProcessor.onNext(SerializedProcessor.java:104)
> 	at io.smallrye.mutiny.subscription.SerializedSubscriber.onItem(SerializedSubscriber.java:74)
> 	at io.smallrye.mutiny.subscription.MultiSubscriber.onNext(MultiSubscriber.java:61)
> 	at io.smallrye.mutiny.operators.multi.MultiRetryWhenOp$RetryWhenOperator.onFailure(MultiRetryWhenOp.java:127)
> 	at io.smallrye.mutiny.subscription.MultiSubscriber.onError(MultiSubscriber.java:73)
> 	at io.smallrye.mutiny.converters.uni.UniToMultiPublisher$UniToMultiSubscription.onFailure(UniToMultiPublisher.java:102)
> 	at io.smallrye.mutiny.operators.uni.UniOperatorProcessor.onFailure(UniOperatorProcessor.java:55)
> 	at io.smallrye.mutiny.vertx.AsyncResultUni.lambda$subscribe$1(AsyncResultUni.java:37)
> 	at io.smallrye.mutiny.vertx.DelegatingHandler.handle(DelegatingHandler.java:25)
> 	at io.vertx.ext.web.client.impl.HttpContext.handleFailure(HttpContext.java:393)
> 	at io.vertx.ext.web.client.impl.HttpContext.execute(HttpContext.java:387)
> 	at io.vertx.ext.web.client.impl.HttpContext.next(HttpContext.java:362)
> 	at io.vertx.ext.web.client.impl.HttpContext.fire(HttpContext.java:329)
> 	at io.vertx.ext.web.client.impl.HttpContext.fail(HttpContext.java:310)
> 	at io.vertx.ext.web.client.impl.HttpContext.lambda$handleCreateRequest$6(HttpContext.java:493)
> 	at io.vertx.core.impl.future.FutureImpl$3.onFailure(FutureImpl.java:153)
> 	at io.vertx.core.impl.future.FutureBase.lambda$emitFailure$1(FutureBase.java:69)
> 	at io.vertx.core.impl.EventLoopContext.execute(EventLoopContext.java:86)
> 	at io.vertx.core.impl.DuplicatedContext.execute(DuplicatedContext.java:163)
> 	at io.vertx.core.impl.future.FutureBase.emitFailure(FutureBase.java:66)
> 	at io.vertx.core.impl.future.FutureImpl.tryFail(FutureImpl.java:230)
> 	at io.vertx.core.impl.future.PromiseImpl.tryFail(PromiseImpl.java:23)
> 	at io.vertx.core.http.impl.HttpClientImpl.lambda$doRequest$6(HttpClientImpl.java:689)
> 	at io.vertx.core.net.impl.pool.Endpoint.lambda$getConnection$0(Endpoint.java:52)
> 	at io.vertx.core.http.impl.SharedClientHttpStreamEndpoint$Request.handle(SharedClientHttpStreamEndpoint.java:162)
> 	at io.vertx.core.http.impl.SharedClientHttpStreamEndpoint$Request.handle(SharedClientHttpStreamEndpoint.java:123)
> 	at io.vertx.core.impl.EventLoopContext.emit(EventLoopContext.java:55)
> 	at io.vertx.core.impl.ContextBase.emit(ContextBase.java:239)
> 	at io.vertx.core.net.impl.pool.SimpleConnectionPool$ConnectFailed$1.run(SimpleConnectionPool.java:384)
> 	at io.vertx.core.net.impl.pool.CombinerExecutor.submit(CombinerExecutor.java:56)
> 	at io.vertx.core.net.impl.pool.SimpleConnectionPool.execute(SimpleConnectionPool.java:245)
> 	at io.vertx.core.net.impl.pool.SimpleConnectionPool.lambda$connect$2(SimpleConnectionPool.java:259)
> 	at io.vertx.core.http.impl.SharedClientHttpStreamEndpoint.lambda$connect$2(SharedClientHttpStreamEndpoint.java:104)
> 	at io.vertx.core.impl.future.FutureImpl$3.onFailure(FutureImpl.java:153)
> 	at io.vertx.core.impl.future.FutureBase.emitFailure(FutureBase.java:75)
> 	at io.vertx.core.impl.future.FutureImpl.tryFail(FutureImpl.java:230)
> 	at io.vertx.core.impl.future.Composition$1.onFailure(Composition.java:66)
> 	at io.vertx.core.impl.future.FutureBase.emitFailure(FutureBase.java:75)
> 	at io.vertx.core.impl.future.FailedFuture.addListener(FailedFuture.java:98)
> 	at io.vertx.core.impl.future.Composition.onFailure(Composition.java:55)
> 	at io.vertx.core.impl.future.FutureBase.emitFailure(FutureBase.java:75)
> 	at io.vertx.core.impl.future.FutureImpl.tryFail(FutureImpl.java:230)
> 	at io.vertx.core.impl.future.PromiseImpl.tryFail(PromiseImpl.java:23)
> 	at io.vertx.core.impl.EventLoopContext.emit(EventLoopContext.java:55)
> 	at io.vertx.core.impl.ContextBase.emit(ContextBase.java:239)
> 	at io.vertx.core.net.impl.NetClientImpl.failed(NetClientImpl.java:347)
> 	at io.vertx.core.net.impl.NetClientImpl.lambda$connectInternal2$6(NetClientImpl.java:319)
> 	at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:590)
> 	at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:557)
> 	at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:492)
> 	at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:636)
> 	at io.netty.util.concurrent.DefaultPromise.setFailure0(DefaultPromise.java:629)
> 	at io.netty.util.concurrent.DefaultPromise.setFailure(DefaultPromise.java:110)
> 	at io.vertx.core.net.impl.ChannelProvider.lambda$handleConnect$0(ChannelProvider.java:157)
> 	at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:590)
> 	at io.netty.util.concurrent.DefaultPromise.notifyListeners0(DefaultPromise.java:583)
> 	at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:559)
> 	at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:492)
> 	at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:636)
> 	at io.netty.util.concurrent.DefaultPromise.setFailure0(DefaultPromise.java:629)
> 	at io.netty.util.concurrent.DefaultPromise.tryFailure(DefaultPromise.java:118)
> 	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.fulfillConnectPromise(AbstractNioChannel.java:321)
> 	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:337)
> 	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:776)
> 	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)
> 	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)
> 	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
> 	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
> 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
> 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
> 	at java.base/java.lang.Thread.run(Unknown Source)
> Caused by: java.lang.RuntimeException: OIDC Server is not available
> 	at io.quarkus.oidc.common.runtime.OidcCommonUtils.lambda$discoverMetadata$3(OidcCommonUtils.java:440)
> 	at io.smallrye.context.impl.wrappers.SlowContextualFunction.apply(SlowContextualFunction.java:21)
> 	at io.smallrye.mutiny.operators.uni.UniOnFailureTransform$UniOnFailureTransformProcessor.onFailure(UniOnFailureTransform.java:54)
> 	... 98 more
> 
> 2024-03-04 08:37:15,375 WARN  [io.qua.oid.com.run.OidcCommonUtils] (vert.x-eventloop-thread-10) OIDC Server is not available:: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: 127.0.0.1.nip.io/127.0.0.1:443
> Caused by: java.net.ConnectException: Connection refused
> 	at java.base/sun.nio.ch.Net.pollConnect(Native Method)
> 	at java.base/sun.nio.ch.Net.pollConnectNow(Unknown Source)
> 	at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(Unknown Source)
> 	at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:337)
> 	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:334)
> 	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:776)
> 	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)
> 	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)
> 	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
> 	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
> 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
> 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
> 	at java.base/java.lang.Thread.run(Unknown Source)`

I've added `127.0.0.1.nip.io` to the `/etc/hosts` file with the IP mapping to `minikube ip`. I'm attempting to access `127.0.0.1.nip.io` from my Windows machine.

Is this error associated with deployment or configurations for theia-cloud?

**Comment by jfaltermeier** [comment]
Hi, if I understand your setup you are running Minikube with Theia Cloud on an Ubuntu machine and are trying to expose this in your network, so you can access this on a windows machine in the same network?
We had a similar question here: https://github.com/eclipsesource/theia-cloud/discussions/232#discussioncomment-6810883

Regarding the error above: `https://127.0.0.1.nip.io/keycloak/realms/TheiaCloud` looks for the keycloak on localhost (and this is probably local to the running docker containers then). This should be `${minikube ip}/keycloak/realms/TheiaCloud` instead.

**Reply by we-wake-act21** [reply]
Hi, 
I resolved this issue by updating the coreDNS configmap to resolve `127.0.0.1.nip.io` to `$(minikube ip)`.

Is this a good approach?
